src/cloudWatchLogs/commands/searchLogGroup.ts

```diff
@@ -99,25 +99,19 @@ export async function searchLogGroup(
     registry: LogDataRegistry,
     logData?: { regionName: string; groupName: string }
 ): Promise<void> {
-    const wizard = new SearchLogGroupWizard(logData)
-
-    const response = await wizard.run()
-
-    let result: Result
-    const source = logData ? 'Explorer' : 'Command'
-
-    if (!response) {
-        result = 'Cancelled'
-        telemetry.cloudwatchlogs_open.emit({ result: result, cloudWatchResourceType: 'logGroup', source: source })
-        return
-    }
-
-    const userResponse = handleWizardResponse(response, registry)
-
-    const uri = createURIFromArgs(userResponse.logGroupInfo, userResponse.parameters)
+    telemetry.cloudwatchlogs_open.run(async span => {
+        const wizard = new SearchLogGroupWizard(logData)
+        telemetry.record({ source: logData ? 'Explorer' : 'Command' })
+        const response = await wizard.run()
+        if (!response) {
+            throw new CancellationError('user')
+        }
 
-    result = await prepareDocument(uri, userResponse, registry)
-    telemetry.cloudwatchlogs_open.emit({ result: result, cloudWatchResourceType: 'logGroup', source: source })
+        const userResponse = handleWizardResponse(response, registry)
+        const uri = createURIFromArgs(userResponse.logGroupInfo, userResponse.parameters)
+        const result = await prepareDocument(uri, userResponse, registry)
+        telemetry.record({ result: result })
```

Comment:
Oh this will lead to unexpected behavior, I didn't implement any logic in `spans.ts` for "overriding" `result`.

The assumption is that, when using `run`, the instrumented code should not (and actually cannot) accurately record the outcome of its own execution.

Reply:
> The assumption is that, when using run, the instrumented code should not (and actually cannot) accurately record the outcome of its own execution.

In this case I assume this means just that `prepareDocument` should either throw or not throw rather than return a `Result`.

Reply:
> In this case I assume this means just that prepareDocument should either throw or not throw rather than return a Result.

Yes sorry this is what I meant, my comment was a little cryptic. If a function cannot proceed it should raise an exception. If callers frequently need to handle those exceptions then using a monad like `Result` can make things more explicit. But either way exceptions are a divergence from the expected behavior of a function.